src/lib/apiMiddlewares/withSurvey.ts

```diff
@@ -9,9 +9,12 @@ export function withSurvey(handler: NextApiHandler) {
   return async function (req: NextApiRequest, res: NextApiResponse) {
     try {
       const session = await getServerSession(req, res, authOptions);
+      if (!session || !session.user) {
+        return res.status(401).end();
+      }
```

Comment:
This is already done in the `withAuth` middleware so no need for it here

You can use it like `withAuth(withSurvey(handler))`